src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/TaskCompletionSource.cs

```diff
@@ -285,5 +285,75 @@ public bool TrySetCanceled(CancellationToken cancellationToken)
+        /// </summary>
+        /// <param name="completedTask">The completed task whose completion status (including exception or cancellation information) should be copied to the underlying task.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="completedTask"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentException"><paramref name="completedTask"/> is not completed.</exception>
+        /// <exception cref="InvalidOperationException">
+        /// The underlying <see cref="Task{TResult}"/> is already in one of the three final states:
+        /// <see cref="TaskStatus.RanToCompletion"/>, <see cref="TaskStatus.Faulted"/>, or <see cref="TaskStatus.Canceled"/>.
+        /// </exception>
+        /// <remarks>
+        /// This operation will return false if the <see cref="Task{TResult}"/> is already in one of the three final states:
+        /// <see cref="TaskStatus.RanToCompletion"/>, <see cref="TaskStatus.Faulted"/>, or <see cref="TaskStatus.Canceled"/>.
+        /// </remarks>
+        public void SetFromTask(Task completedTask)
+        {
+            if (!TrySetFromTask(completedTask))
+            {
+                ThrowHelper.ThrowInvalidOperationException(ExceptionResource.TaskT_TransitionToFinal_AlreadyCompleted);
+            }
+        }
+
+        /// <summary>
+        /// Attempts to transition the underlying <see cref="Task{TResult}"/> into the same completion state as the specified <paramref name="completedTask"/>.
+        /// </summary>
+        /// <param name="completedTask">The completed task whose completion status (including exception or cancellation information) should be copied to the underlying task.</param>
+        /// <returns><see langword="true"/> if the operation was successful; otherwise, <see langword="false"/>.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="completedTask"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentException"><paramref name="completedTask"/> is not completed.</exception>
+        /// <remarks>
+        /// This operation will return false if the <see cref="Task{TResult}"/> is already in one of the three final states:
+        /// <see cref="TaskStatus.RanToCompletion"/>, <see cref="TaskStatus.Faulted"/>, or <see cref="TaskStatus.Canceled"/>.
+        /// </remarks>
+        public bool TrySetFromTask(Task completedTask)
+        {
+            ArgumentNullException.ThrowIfNull(completedTask);
+            if (!completedTask.IsCompleted)
+            {
+                throw new ArgumentException(SR.Task_MustBeCompleted, nameof(completedTask));
+            }
+
+            // Try to transition to the appropriate final state based on the state of completedTask.
+            bool result = false;
+            switch (completedTask.Status)
+            {
+                case TaskStatus.RanToCompletion:
+                    result = _task.TrySetResult();
```

Comment:
is there test for scenario in which `completedTask` is `Task<T>`. I guess it should work just fine.

Reply:
I didn't add one for this specific API as it doesn't try to use anything on any derived Task type.